src/integrations/mailchimp/class-mailchimp.php

```diff
@@ -0,0 +1,203 @@
+ *
+ * @package Eightshift_Forms\Integrations
+ */
+
+declare( strict_types=1 );
+
+namespace Eightshift_Forms\Integrations\Mailchimp;
+
+use Eightshift_Forms\Hooks\Filters;
+use Eightshift_Forms\Integrations\Core\Http_Client;
+use Eightshift_Forms\Exception\Missing_Filter_Info_Exception;
+use \MailchimpMarketing\ApiClient as MarketingApiClient;
+/**
+ * Mailchimp integration class.
+ */
+class Mailchimp {
+
+  /**
+   * Mailchimp Marketing Api client.
+   *
+   * @var ApiClient
+   */
+  private $client;
+
+  /**
+   * Constructs object
+   *
+   * @param Http_Client $guzzle_client Guzzle client.
+   */
+  public function __construct( Http_Client $guzzle_client ) {
+    $this->guzzle_client = $guzzle_client;
+  }
+
+  /**
+   * Adds or updates a member in Mailchimp.
+   *
+   * @param  string $list_id      Audience list ID.
+   * @param  string $email        Contact's email.
+   * @param  array  $merge_fields List of merge fields to add to request.
+   * @param  array  $params       (Optional) list of params from request.
+   * @param  string $status       (Optional) Member's status (if new).
+   * @return mixed
+   *
+   * @throws \Exception When response is invalid.
+   */
+  public function add_or_update_member( string $list_id, string $email, array $merge_fields, array $params = [], string $status = 'pending' ) {
+    $this->maybe_build_client();
+
+    $params['email_address'] = $email;
+    $params['status_if_new'] = $status;
+    $params['merge_fields']  = $merge_fields;
+
+    $response = $this->client->lists->setListMember( $list_id, $this->calculate_subscriber_hash( $email ), $params );
+
+    if ( ! is_object( $response ) || ! isset( $response->id, $response->email_address ) ) {
+      throw new \Exception( 'setListMember response invalid' );
+    }
+
+    return $response;
+  }
+
+  /**
+   * Add a tag to a member.
+   *
+   * @param  string $list_id   Audience list ID.
+   * @param  string $email     Contact's email.
+   * @param  array  $tag_names Just a 1d array of tag names. Other processing is done inside.
+   * @return bool
+   *
+   * @throws \Exception When response is invalid.
+   */
+  public function add_member_tags( string $list_id, string $email, array $tag_names ): bool {
+    $this->maybe_build_client();
+
+    // This call requires a very specific format for tags.
+    $tags_request = [
+      'tags' => array_map(function( $tag_name ) {
+          return [
+            'name' => $tag_name,
+            'status' => 'active',
+          ];
+      }, $tag_names),
+    ];
+
+    $update_response = $this->client->lists->updateListMemberTags( $list_id, md5( $email ), $tags_request );
+
+    // This call is weird in that an empty (204) response means success. If something went very wrong it
+    // will throw an exception. If something is slightly off (such as not having the correct format for
+    // tags array), it will also return an empty response.
+    return ! $update_response ? true : false;
```

Comment:
What is the actual value of `$update_response`? Because this looks like an unnecessary ternary.

Reply:
Yup, unnecessary ternary. Also now I'm checking with `empty()` because that's more correct.